freeqdsk/aeqdsk.py

```diff
@@ -1,280 +1,582 @@
+    Field("zvsiu", description="Z of inner upper strikepoint", default=0.0),
+    Field("rvsid", description="major radius of inner lower strikepoint", default=0.0),
+    Field("zvsid", description="Z of inner lower strikepoint", default=0.0),
+    Field("rvsou", description="major radius of outer upper strikepoint", default=0.0),
+    Field("zvsou", description="Z of outer upper strike point", default=0.0),
+    Field("rvsod", description="major radius of outer lower strike point", default=0.0),
+    Field("zvsod", description="Z of outer lower strike point", default=0.0),
+    Field(
+        "condno",
+        description="condition number from least-squares fitting routine",
+        default=0.0,
+    ),
+    Field(
+        "dollbaf", description="distance from outer leg to lower baffle", default=0.0
+    ),
+    Field(
+        "dollbafm",
+        description=(
+            "distance from outer midplane last closed flux surface to the flux surface "
+            "intersecting the lower baffle"
+        ),
+        default=0.0,
+    ),
+    Field("dilldom", description="distance from inner leg to lower dome", default=0.0),
+    Field(
+        "dilldomm",
+        description=(
+            "distance from inner midplane last closed flux surface and the flux "
+            "surface intersecting the lower dome"
+        ),
+        default=0.0,
+    ),
+    Field("dummy_1", description="Purpose unknown", default=0.0),
+    Field("dummy_2", description="Purpose unknown", default=0.0),
 ]
 
 
-def write(data, fh):
+def fields() -> Dict[str, Field]:
+    r"""
+    Returns full list of all Fields.
     """
-    data  [dict] - keys are given with documentation in the `fields` list.
-        Also includes
-         shot [int] - The shot number
-    time  - in ms
+    all_fields = (
+        _general_block_1
+        + _laser_block
+        + _general_block_2
+        + _extended_sizes
+        + _extended_arrays_1
+        + _extended_arrays_2
+        + _extended_general
+    )
+    return {field.name: field for field in all_fields}
 
+
+def _float_field(field: Field, data: Dict[str, Union[float, int, ArrayLike]]) -> float:
+    """
+    Returns single floating point field data from dict if present.
+    Otherwise returns a default.
     """
+    return float(data.get(field.name, field.default))
+
+
+def _len_field(field: Field, data: Dict[str, Union[float, int, ArrayLike]]) -> int:
+    """
+    Returns length field data from dict if present.
+    Otherwise gets length from array (if present) or returns 0.
+    """
+    if field.name in data:
+        # Check length of array, if present
+        if field.length_of in data and data[field.name] != len(data[field.length_of]):
+            raise ValueError(
+                f"'{field.name}' should be the length of the array '{field.length_of}'"
+            )
+        return int(data[field.name])
+    else:
+        return len(data.get(field.length_of, []))
```

Comment:
Move the guard clause to the top:

```suggestion
    if field.name not in data:
        return len(data.get(field.length_of, []))

    # Check length of array, if present
    if field.length_of in data and data[field.name] != len(data[field.length_of]):
        raise ValueError(
            f"'{field.name}' should be the length of the array '{field.length_of}'"
        )
    return int(data[field.name])
```